Configurations/shared-info.pl

```diff
@@ -0,0 +1,108 @@
+#! /usr/bin/env perl
+# -*- mode: perl; -*-
+# Copyright 2016-2017 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+# This is a collection of extra attributes to be used as input for creating
+# shared libraries, currently on any Unix variant, including Unix like
+# environments on Windows.
+
+sub detect_gnu_ld {
+    my @lines =
+        `$config{cross_compile_prefix}$target{cc} -Wl,-V /dev/null 2>&1`;
+    return grep /^GNU ld/, @lines;
+}
+sub detect_gnu_cc {
+    my @lines =
+        `$config{cross_compile_prefix}$target{cc} -v 2>&1`;
+    return grep /gcc/, @lines;
+}
+
+my %shared_info;
```

Comment:
combine the two lines?  And with that, my Perl expertise is all used up :)

Reply:
Good point

Reply:
(I think...  since I have references to `%shared_info` from within the subs in this structure, I'm worried that combining will mean the symbol isn't lexigraphically defined when those subs are compiled.

Reply:
Ahyup, when I combined into one line, I got this:

```
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 32.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 36.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 38.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 54.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 61.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 68.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 75.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 82.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 89.
Global symbol "%shared_info" requires explicit package name (did you forget to declare "my %shared_info"?) at (eval 11) line 96.
```

So, errr, no.